pq/util.go

```diff
@@ -28,23 +28,14 @@ func getPage(tx *txfile.Tx, id txfile.PageID) ([]byte, error) {
 	return page.Bytes()
 }
 
-func withPage(tx *txfile.Tx, id txfile.PageID, fn func([]byte) error) error {
+func withPage(tx *txfile.Tx, id txfile.PageID, fn func([]byte)) error {
 	page, err := getPage(tx, id)
 	if err != nil {
 		return err
 	}
-	return fn(page)
-}
-
-func readPageByID(accessor *access, pool *pagePool, id txfile.PageID) (*page, error) {
-	tx := accessor.BeginRead()
-	defer tx.Close()
 
-	var page *page
-	return page, withPage(tx, id, func(buf []byte) error {
-		page = pool.NewPageWith(id, buf)
-		return nil
-	})
+	fn(page)
```

Comment:
Did not fully understand why `fn` does not return an error anymore.

Reply:
We have 2 error types. Plus there was only one function ever reporting an error. This function does not return an error anymore, but stores the error in a local variable. By not returning an error anymore and capturing it I can distinguish where the error actually occured + I don't have to deal with different error types.